components/bl-leaflet-map/src/helpers/map.js

```diff
@@ -86,21 +86,35 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
   });
 }
 
+function isCircleDataValid(numbers, text, radius) {
+  const isNumbers = numbers.every(value => !isNaN(value));
+  const isText = text.every(value => typeof value === "string");
+  const isRadius = radius > 0;
+
+  return isNumbers && isText && isRadius;
+}
+
 export function createCircles(circles, map, eventHandlers) {
   if (circles) {
     const { onCircleClick } = eventHandlers;
 
     circles.forEach(item => {
-      Leaflet.circle([item.point.lat, item.point.lng], { radius: item.radius })
-        .on('click', () => {
-          onCircleClick({
-            coordinates: [item.point.lat, item.point.lng],
-            radius     : item.radius,
-            description: item.description,
-          });
-        })
-        .addTo(map)
-        .bindPopup(item.description);
+      const { point: { lat, lng }, radius, description } = item;
+
+      if (isCircleDataValid([lat, lng, radius], [description], radius)) {
+        Leaflet.circle([lat, lng], { radius })
+          .on('click', () => {
+            onCircleClick({
+              coordinates: [lat, lng],
+              radius     : radius,
+              description: description,
+            });
+          })
+          .addTo(map)
+          .bindPopup(description);
+      } else {
+        console.warn('Circle data is not valid!');
```

Comment:
That looks like an error. Also, need extend error message with invalidation reason